### PR TITLE
[MCMC 1/3] Moving clifford utils for cdr to own files

### PR DESCRIPTION
**Pull request by isaac-gs**
## Description
This pull request is one of several parts involved in #1399. The refactoring is necessary to improve the dev experience moving forward.

---

### License

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [x] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: mitiq/cdr/clifford_training_data.py [hunk 1]
```diff
@@ -43,14 +42,11 @@ def generate_training_circuits(
...
     non-Clifford gates in the input circuit by Clifford gates.
-
     The way in which non-Clifford gates are selected to be replaced is
     determined by ``method_select`` and ``method_replace``.
-
     In the Clifford Data Regression (CDR) method
     :cite:`Czarnik_2021_Quantum`, data generated from these circuits is used
     as a training set to learn the effect of noise.
-
     Args:
```
Comment (andreamari) [inline mitiq/cdr/clifford_training_data.py:50]: Docs tests are failing in the GitHub CI tests. Not sure why. Maybe an empty line is necessary before `Args:`.

Comment (andreamari) [reply mitiq/cdr/clifford_training_data.py:50]: btw, thanks for this PR! I Iike the new file structure

Comment (isaac-gs) [reply mitiq/cdr/clifford_training_data.py]: Done, no problem. First of several. Before I merge I'll let you see the new complete structure

**Comment by isaac-gs** [comment]
@andreamari This is part 2, I wanted to setup the PRs as being stacked to make reviewing the changes easier but it looks like forking the repo makes that difficult. https://github.com/unitaryfund/mitiq/pull/1485

**Comment by isaac-gs** [comment]
btw, this test passes for me locally :(

mitiq/zne/tests/test_inference.py

**Comment by andreamari** [comment]
> btw, this test passes for me locally :(
> 
> mitiq/zne/tests/test_inference.py

The problem seems unrelated to this PR. I am re-running tests to see if they pass now.

**Comment by andreamari** [comment]
> @andreamari This is part 2, I wanted to setup the PRs as being stacked to make reviewing the changes easier but it looks like forking the repo makes that difficult. #1485

If we merge this PR (MCMC 1/3), is it going to help? For me this is ready to go, as soon as tests pass (we can ignore codecov)

**Comment by isaac-gs** [comment]
@andreamari yeah we can go ahead and merge

**Review by andreamari — APPROVED** [review]
LGTM, thanks @isaac-gs !

**Pull request merged**